### PR TITLE
Fix em_cpuhotplug test

### PR DESCRIPTION
**Pull request by PraveenPenguin**
1-Fix: few pylint issues 
2-cleanup and use Avocado library 

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

File: cpu/em_cpuhotplug.py [hunk 1]
```diff
@@ -14,14 +14,15 @@
...
 # Author: Gautham R. Shenoy <ego@linux.vnet.ibm.com>
 # Author: Shriya Kulkarni <shriyak@linux.vnet.ibm.com>
+
 import random
-import platform
+
 from avocado import Test
 from avocado import main
-from avocado.utils import process, cpu
+from avocado.utils import process, cpu, genioi, distro
```
Comment (narasimhan-v) [inline cpu/em_cpuhotplug.py]: genio

File: cpu/em_cpuhotplug.py [hunk 2]
```diff
@@ -33,8 +34,9 @@ def setUp(self):
         Get the number of cores and threads per core
         Set the SMT value to 4/8
         """
-        if 'ppc' not in platform.processor():
-            self.cancel("Processor is not ppc64")
+        if distro.detect().arch in ['ppc64', 'ppc64le']:
```
Comment (narasimhan-v) [inline cpu/em_cpuhotplug.py]: not in

File: cpu/em_cpuhotplug.py [hunk 3]
```diff
@@ -46,23 +48,23 @@ def setUp(self):
...
-                       ignore_status=True)
+
+        genio.write_one_line('/proc/sys/kernel/printk', "8")
         self.max_smt = 4
         if cpu.get_cpu_arch().lower() == 'power8':
             self.max_smt = 8
         if cpu.get_cpu_arch().lower() == 'power6':
             self.max_smt = 2
-        process.system_output("ppc64_cpu --smt=%s" % self.max_smt, shell=True)
+        process.system("ppc64_cpu --smt=%s" % self.max_smt, shell=True)
```
Comment (narasimhan-v) [inline cpu/em_cpuhotplug.py]: what is the reason for change ?

**Comment by narasimhan-v** [comment]
@PraveenPenguin attach job.log also.

File: cpu/em_cpuhotplug.py (hunk 3 above)
Comment (PraveenPenguin) [reply cpu/em_cpuhotplug.py]: as system_output is not needed here as we not capturing it any where

File: cpu/em_cpuhotplug.py (hunk 2 above)
Comment (PraveenPenguin) [reply cpu/em_cpuhotplug.py]: ah sure

File: cpu/em_cpuhotplug.py (hunk 1 above)
Comment (PraveenPenguin) [reply cpu/em_cpuhotplug.py]: my bad :(

**Comment by PraveenPenguin** [comment]
Job Log:
```[root@system~]# avocado run em_cpuhotplug.py -m em_cpuhotplug.py.data/em_cpuhotplug.yaml 
JOB ID     : 706b971ad0d214c64bde0eb025e30b9b451537d1
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2019-07-04T05.33-706b971/job.log
 (1/1) em_cpuhotplug.py:Cpuhotplug_Test.test;run-52ff: PASS (482.50 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 482.97 s
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2019-07-04T05.33-706b971/results.html
[root@system ~]# 
````

**Comment by PraveenPenguin** [comment]
@narasimhan-v pushed some more commit , added job log too please check

File: cpu/em_cpuhotplug.py [hunk 4]
```diff
@@ -37,6 +37,7 @@ def setUp(self):
         if distro.detect().arch not in ['ppc64', 'ppc64le']:
             self.cancel("Only supported in powerpc system")
 
+        self.loop = self.params.get('test_loop', default=100)
```
Comment (narasimhan-v) [inline cpu/em_cpuhotplug.py]: int(self.params.get('test_loop', default=100))

Comment (PraveenPenguin) [reply cpu/em_cpuhotplug.py]: ok sure makes sense if user had string in yaml

**Comment by PraveenPenguin** [comment]
@narasimhan-v pushed the changes please have a look

**Pull request merged**